mat64/shadow_test.go

```diff
@@ -115,3 +258,27 @@ func TestIssue359(t *testing.T) {
 		}
 	}
 }
+
+// denseAsTriDense returns a triangular matrix derived from the
+// square matrix m, with the orientation specified by kind.
+func denseAsTriDense(m *Dense, kind matrix.TriKind) *TriDense {
```

Comment:
I'm leading the way in bad behaviour because that's what we are trying to catch.